chalk-engine/src/table.rs

```diff
@@ -72,8 +68,18 @@ impl<C: Context> Table<C> {
         mem::replace(&mut self.strands, VecDeque::new())
     }
 
-    pub(crate) fn pop_next_strand(&mut self) -> Option<Strand<C>> {
-        self.strands.pop_front()
+    pub(crate) fn pop_next_strand_if(
+        &mut self,
+        test: impl Fn(&Strand<C>) -> bool,
+    ) -> Option<Strand<C>> {
+        let strand = self.strands.pop_front();
+        if let Some(strand) = strand {
+            if test(&strand) {
+                return Some(strand);
+            }
+            self.strands.push_front(strand);
```

Comment:
isn't this going to just test and re-test the same strand over and over?

Reply:
So, it would, but the assumption is that any `Strand`s after this have a `>= work`.